core/src/banking_stage/unprocessed_transaction_storage.rs

```diff
@@ -149,8 +154,16 @@ fn consume_scan_should_process_packet(
         return ProcessingDecision::Now;
     }
 
-    // Before sanitization, let's quickly check the static keys (performance optimization)
+    // Check if the fee payer is in the invalid fee payer filter
     let message = &packet.transaction().get_message().message;
+    if invalid_fee_payer_filter.should_reject(&message.static_account_keys()[0]) {
```

Comment:
Check for known invalid fee-payers during MI batch selection. Don't even attempt to load and execute these transactions.